pkglistgen/tool.py

```diff
@@ -620,67 +681,101 @@ def update_and_solve_target(
+          if len(root.xpath('result[@state="building"]')) or len(root.xpath('result[@state="dirty"]')):
+              logging.info('{}/{} build in progress'.format(project, product))
+              return
+        if git_url:
+            if os.path.exists(cache_dir + "/.git"):
+                # reset and update existing clone
+                logging.debug(subprocess.check_output(
+                    ['git', 'reset', '--hard'], cwd=cache_dir, encoding='utf-8'))
+                logging.debug(subprocess.check_output(
+                    ['git', 'pull', '--rebase'], cwd=cache_dir, encoding='utf-8'))
+            else:
+                if os.path.exists(cache_dir):
+                    shutil.rmtree(cache_dir)
+                logging.debug(subprocess.check_output(
+                    ['git', 'clone', '--recurse-submodules', git_url, cache_dir], encoding='utf-8'))
+            if os.path.exists(cache_dir + '/' + '000update-repos'):
+                logging.error('No support for 000update-repos in git projects atm')
+                return
+        else:
+            url = api.makeurl(['source', project])
+            packages = ET.parse(http_GET(url)).getroot()
+            if packages.find('entry[@name="{}"]'.format(product)) is None:
+                if not self.dry_run:
+                    undelete_package(api.apiurl, project, product, 'revive')
+                # TODO disable build.
+                logging.info('{} undeleted, skip dvd until next cycle'.format(product))
+                return
 
-        if packages.find('entry[@name="{}"]'.format(release)) is None:
-            if not self.dry_run:
-                undelete_package(api.apiurl, project, release, 'revive')
-            logging.info('{} undeleted, skip dvd until next cycle'.format(release))
-            return
+            drop_list = api.item_exists(project, oldrepos)
+            if drop_list and not only_release_packages:
+                checkout_list.append(oldrepos)
 
-        # Cache dir specific to hostname and project.
-        host = urlparse(api.apiurl).hostname
-        cache_dir = CacheManager.directory('pkglistgen', host, project)
+            if packages.find('entry[@name="{}"]'.format(release)) is None:
+                if not self.dry_run:
+                    undelete_package(api.apiurl, project, release, 'revive')
+                logging.info('{} undeleted, skip dvd until next cycle'.format(release))
+                return
 
-        if not no_checkout:
-            if os.path.exists(cache_dir):
-                shutil.rmtree(cache_dir)
-            os.makedirs(cache_dir)
+            if not no_checkout:
+                if os.path.exists(cache_dir):
+                    shutil.rmtree(cache_dir)
+                os.makedirs(cache_dir)
 
         group_dir = os.path.join(cache_dir, group)
         product_dir = os.path.join(cache_dir, product)
+        productcompose_dir = self.productcompose_dir = os.path.join(cache_dir, productcompose)
```

Comment:
Should use either `self.productcompose_dir` or `productcompose_dir`, not both

Reply:
done